ext/standard/math.c

```diff
@@ -120,13 +120,35 @@ static inline double php_round_helper(double value, int mode) {
 			return integral;
 
 		case PHP_ROUND_HALF_DOWN:
+			fractional = fabs(modf(value, &integral));
 			if (fractional > 0.5) {
 				return integral + copysign(1.0, integral);
 			}
 
 			return integral;
 
+		case PHP_ROUND_CEILING:
+			return ceil(value);
```

Comment:
Would this be `fractional == 0 ? integral : integral + 1.0` or is there a case where this would give a different result?